taf/updater/lifecycle_handlers.py

```diff
@@ -282,6 +293,67 @@ def prepare_data_host(
     return host_data_by_repos
 
 
+def prepare_data_update(
+    event,
+    transient_data,
+    persistent_data,
+    library_dir,
+    hosts,
+    repos_update_data,
+    error,
+    root_auth_repo,
+):
+
+    update_hosts = {}
+    flat_auth_repos = {}
+    for auth_repo_name in repos_update_data.keys():
+        flat_auth_repos[auth_repo_name] = _repo_update_data(
+            **repos_update_data[auth_repo_name]
+        )
+
+    for host_data in hosts:
+        auth_repos = {}
+        for root_repo, contained_repo_data in host_data.data_by_auth_repo.items():
+            for host_auth_repos in contained_repo_data["auth_repos"]:
+                for repo_name, repo_host_data in host_auth_repos.items():
+                    auth_repo = {}
+                    auth_repo[repo_name] = _repo_update_data(
+                        **repos_update_data[repo_name]
+                    )
+                    auth_repo[repo_name]["custom"] = repo_host_data["custom"]
+                    auth_repos.update({"update": auth_repo})
+
+            update_hosts[host_data.name] = {
+                "host_name": host_data.name,
+                "error_msg": str(error) if error else "",
+                "auth_repos": auth_repos,
+                "custom": contained_repo_data["custom"],
+            }
+
+    return {
+        root_auth_repo: {
+            "data": {
+                "update": {
+                    "changed": event == Event.CHANGED,
+                    "event": _format_event(event),
+                    "error_msg": str(error) if error else "",
+                    "hosts": update_hosts,
+                    "auth_repos": flat_auth_repos,
+                    "auth_repo": root_auth_repo.name,
```

Comment:
Not sure if `auth_repo_name` is more intuitive  - if `auth_repos` contains the objects, I'd expect `auth_repo` to be one repo object

Reply:
makes sense. so entire root_auth_repo object instead of the name?